idea/src/org/jetbrains/kotlin/idea/intentions/AddLabeledReturnInLambdaIntention.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.codeInsight.intention.LowPriorityAction
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.getCallNameExpression
+import org.jetbrains.kotlin.psi.psiUtil.getNonStrictParentOfType
+import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.types.typeUtil.isUnit
+
+class AddLabeledReturnInLambdaIntention :
+        SelfTargetingIntention<KtExpression>(KtExpression::class.java, "Add labeled return to last expression in a lambda"),
+        LowPriorityAction {
+    override fun isApplicableTo(element: KtExpression, caretOffset: Int): Boolean {
+        if (element.getNonStrictParentOfType<KtReturnExpression>() != null) return false
+        val block = element.getStrictParentOfType<KtBlockExpression>() ?: return false
+        if (block.statements.lastOrNull() != element) return false
+        val callExpression = element.getStrictParentOfType<KtCallExpression>() ?: return false
+        val index = callExpression.valueArguments.indexOfFirst {
+            val lambda = it.getArgumentExpression() as? KtLambdaExpression
+            lambda?.bodyExpression == block
+        }
+        if (index < 0) return false
+        val resolvedCall = callExpression.getResolvedCall(callExpression.analyze()) ?: return false
+        if (resolvedCall.candidateDescriptor.valueParameters[index].type.arguments.last().type.isUnit()) return false
+        val labelName = callExpression.getCallNameExpression()?.text ?: return false
```

Comment:
What if the lambda is labeled? E.g.

```
    listOf(1, 2).find label@{
        true
    }
```

Reply:
Wow. I didn't know there is such syntax in Kotlin. Let me re-think about this.